lib/API/JSON/v1/Subscribers.php

```diff
@@ -95,24 +95,42 @@ function listing($data = array()) {
       $listing_data = $this->subscribers_listings->getListingsInSegment($data);
     }
 
-    $data = array();
+    $result = array();
     foreach ($listing_data['items'] as $subscriber) {
-      $data[] = $subscriber
+      $subscriber_result = $subscriber
         ->withSubscriptions()
         ->asArray();
+      if (isset($data['filter']['segment'])) {
+        $subscriber_result = $this->fixSubscriberStatus($subscriber_result, $data['filter']['segment']);
+      }
+      $result[] = $subscriber_result;
     }
 
     $listing_data['filters']['segment'] = $this->wp->applyFilters(
       'mailpoet_subscribers_listings_filters_segments',
       $listing_data['filters']['segment']
     );
-    return $this->successResponse($data, array(
+
+    return $this->successResponse($result, array(
       'count' => $listing_data['count'],
       'filters' => $listing_data['filters'],
       'groups' => $listing_data['groups']
     ));
   }
 
+  private function fixSubscriberStatus(array $subscriber, $segment_id) {
+    $segment_status = Subscriber::STATUS_SUBSCRIBED;
```

Comment:
Why is this set to `subscribed`? It seems unnecessary.

Reply:
The line is not necessary, but it is here to make sure the variable is always defined and always the same type. But maybe it is confusing and I'll refactor the function a bit.